### PR TITLE
MAGIC variance reduction

### PR DESCRIPTION
**Pull request by kkiesling**
I added the MAGIC variance reduction technique to variancereduction.py. @makeclean, you should review the method. Tests were also added.

Theory manual addition still needed.

issue #344

### CODE REVIEW

**Comment by scopatz** [comment]
Looks like this could use a rebase | merge from develop

**Comment by scopatz** [comment]
Hey @kkiesling - now that this passes CI and is mergeable, is this ready for another review (and merge?)?

**Comment by scopatz** [comment]
I think that this would be good to have for v0.5, if it can make it in there.

**Comment by kkiesling** [comment]
If you are okay with merging it without the documentation, then I believe it is ready. I can't make any promises to have the documentation written until after March 1st unfortunately but it will be done in that week if I can't get it done earlier.

**Comment by scopatz** [comment]
Hi @kkiesling - I think we need documentation.  Why don't we leave this in v0.5 for now, and if it happens to be the last remaining v0.5 issue open we'll make a decision then?

File: pyne/variancereduction.py [hunk 1]
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
...
                          for i in range(num_e_groups)]
+
+
+def magic(meshtally, tag_name, tag_name_error, **kwargs):
+    """This function reads a PyNE mcnp.MeshTally object and preforms the MAGIC 
+    algorithm and returns the resulting weight window mesh.
+
+    Parameters:
+    -----------
+        meshtally : a single PyNE mcnp.MeshTally obj
```
Comment (scopatz) [inline pyne/variancereduction.py]: Unindent these lines please to followi the numpydoc rules.

File: pyne/variancereduction.py [hunk 2]
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
...
+            create a weight window lower bound for for a given mesh volume 
+            element for the intial weight window lower bound generation, or 
+            overwrite preexisting weight window lower bounds for subsequent 
+            iterations. 
+        null_value : float, optional
+            The weight window lower bound value that is assigned to mesh volume
+            elements where the relative error on flux exceeds the tolerance.
+    """
+    
+    tolerance = kwargs["tolerance"] if "tolerance" in kwargs else 0.5
```
Comment (scopatz) [inline pyne/variancereduction.py]: this is better as 

```python
tolerace = kwargs.get('tolerance', 0.5)
```

and similarly below.

File: pyne/variancereduction.py [hunk 3]
```diff
@@ -126,3 +128,103 @@ def cadis(adj_flux_mesh, adj_flux_tag, q_mesh, q_tag,
...
+    null_value = kwargs["null_value"] if "null_value" in kwargs else 0.0
+    
+    # Convert particle name to the recognized abbreviation
+    particle = (meshtally.particle[0].upper() + meshtally.particle[1:])
```
Comment (scopatz) [inline pyne/variancereduction.py]: you probably want to do `meshtally.capitalize()` here.

**Comment by scopatz** [comment]
couple of minor style issues @kkiesling, otherwise it loosk good to go.

**Comment by kkiesling** [comment]
Pinging this. I fixed the style issues, finished the manual entry, and added the test that @makeclean was talking about a while back. This should be good to go officially when Andy approves.

**Comment by scopatz** [comment]
Looks good to me, @makeclean?

**Comment by makeclean** [comment]
Yup yup. Merging.

**Pull request merged**